packages/@aws-cdk/aws-cloudfront-origins/lib/rest-api-origin.ts

```diff
@@ -6,7 +6,7 @@ import { validateSecondsInRangeOrUndefined } from './private/utils';
 /**
  * Properties for an Origin for an API Gateway REST API.
  */
-export interface RestApiOriginProps extends cloudfront.OriginOptions {
+export interface RestApiOriginProps extends cloudfront.OriginProps {
```

Comment:
What is in OriginProps that isn't in OriginOptions and vice versa? I'm concerned this might break other languages even if it's fine in TypeScript.

Reply:
1) `OriginProps` extends `OriginOptions` and declares the `originPath`.

```ts
export interface OriginOptions {
  readonly connectionTimeout?: Duration;
  readonly connectionAttempts?: number;
  readonly customHeaders?: Record<string, string>;
  readonly originShieldRegion?: string;
  readonly originShieldEnabled?: boolean;
  readonly originId?: string;
}

export interface OriginProps extends OriginOptions {
  readonly originPath?: string;
}
```

2) `OriginProps` is used in other packages. Where `OriginOptions` is internal? (not sure)

![Screenshot 2023-02-10 at 11 52 31 PM](https://user-images.githubusercontent.com/829902/218075167-a529e64f-e4a2-47b8-a654-c2390be3178e.png)

![Screenshot 2023-02-10 at 11 57 52 PM](https://user-images.githubusercontent.com/829902/218075557-87d6b601-38c5-4362-9fdf-bd6b1deea910.png)

![Screenshot 2023-02-10 at 11 58 10 PM](https://user-images.githubusercontent.com/829902/218075570-178713f2-a6ed-4951-bf3c-f617b8cff624.png)

Reply:
since the only new property is optional, this should be non-breaking